### PR TITLE
CRI: rearrange kubelet rutnime initialization

### PR DESCRIPTION
**Pull request by yujuhong**
Consolidate the code used by docker+cri and remote+cri for consistency, and to
prevent changing one without the other.  Enforce that
`--experimental-runtime-integration-type` has to be set in order for kubelet
use the CRI interface, *even for out-of-process shims`. This simplifies the
temporary `if` logic in kubelet while CRI still co-exists with older logic.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36201)

### CODE REVIEW

**Comment by feiskyer** [comment]
nit: typo in title

**Comment by Random-Liu** [comment]
LGTM

**Comment by Random-Liu** [comment]
@k8s-bot cri test this.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -253,6 +253,18 @@ func makePodSourceConfig(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps
 	return cfg, nil
 }
 
+func getRuntimeAndImageServices(config *componentconfig.KubeletConfiguration) (internalApi.RuntimeService, internalApi.ImageManagerService, error) {
+	rs, err := remote.NewRemoteRuntimeService(config.RemoteRuntimeEndpoint, config.RuntimeRequestTimeout.Duration)
+	if err != nil {
+		return nil, nil, err
+	}
+	is, err := remote.NewRemoteImageService(config.RemoteRuntimeEndpoint, config.RuntimeRequestTimeout.Duration)
```
Comment (feiskyer) [inline pkg/kubelet/kubelet.go]: s/RemoteRuntimeEndpoint/RemoteImageEndpoint

**Comment by yujuhong** [comment]
Applied lgtm based on https://github.com/kubernetes/kubernetes/pull/36201#issuecomment-258321468. The cri tests already passed once, so there shouldn't be any surprises.

**Comment by feiskyer** [comment]
LGTM

**Comment by feiskyer** [comment]
@k8s-bot unit test this

**Comment by yujuhong** [comment]
Rebased to resolve conflict with #36020.

**Comment by yujuhong** [comment]
@k8s-bot cri test this

**Comment by yujuhong** [comment]
@k8s-bot cri test this

**Comment by yujuhong** [comment]
Marked p2 to resolve PR dependencies

**Pull request merged**